training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -46,12 +46,32 @@ def __init__(self, model_id: str):
         self._model_uri = self.HF_ENDPOINT + f"/{model_id}"
         self._model_info = None
         self._hf_api = HfApi(endpoint=self.HF_ENDPOINT)
+        self.is_valid_id = None
+
+    @property
+    def is_valid_id(self):
+        """Returns true if Hugging face model id is valid."""
+        return self._is_valid_id
+
+    @is_valid_id.setter
+    def is_valid_id(self, value):
+        """Validate Hugging face model id."""
+        if not value:
+            response = requests.get(self._model_uri)
+            if response.status_code == 404:
+                error_msg = f"Invalid Hugging face model id: {self._model_id}.\
+                      Please ensure that you are using a correct and existing model ID"
+                raise ValueError(error_msg)
+            else:
+                self.is_valid_id = True
```

Comment:
What happens if the request fails and it's not a 404?

Reply:
Changed the approach using hfapi now to validate the models. If we will get empty list for model name we will mark it is as invalid id